vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/bdd/steps/integration/ResourceCheckSteps.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.bdd.steps.integration;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.jbehave.core.annotations.Then;
+import org.jbehave.core.model.ExamplesTable;
+import org.jsoup.nodes.Element;
+import org.vividus.bdd.steps.api.ApiSteps;
+import org.vividus.http.HttpMethod;
+import org.vividus.http.HttpTestContext;
+import org.vividus.http.client.HttpResponse;
+import org.vividus.reporter.event.AttachmentPublisher;
+import org.vividus.softassert.SoftAssert;
+import org.vividus.testcontext.ContextCopyingExecutor;
+import org.vividus.ui.web.configuration.WebApplicationConfiguration;
+import org.vividus.util.HtmlUtils;
+import org.vividus.util.UriUtils;
+import org.vividus.validator.ResourceValidator;
+import org.vividus.validator.model.CheckStatus;
+import org.vividus.validator.model.ResourceValidation;
+
+public class ResourceCheckSteps
+{
+    private static final Set<String> ALLOWED_SCHEMES = Set.of("http", "https");
+    private static final String EXCLUDE_PATTERN = "#|javascript:void\\(0\\)|mailto:.+";
```

Comment:
I think if allowed schemes are defined, then URLs like `mailto:aa@aa.com` and `javascript:void(0)` will be filtered out because their schemes `mailto` and `javascript` will not be in the allowed set

Reply:
You're right